vendors/infineon/secure_elements/pkcs11/iot_pkcs11_trustx.c

```diff
@@ -2265,8 +2306,15 @@ CK_DEFINE_FUNCTION( CK_RV, C_FindObjects )( CK_SESSION_HANDLE xSession,
 
                 PKCS11_PAL_GetObjectValueCleanup( pcObjectValue, xObjectLength );
             }
+            else
+            {
+                /* The object handle is recognized but its contents don't exist
+                 * in storage. */
+                *pxObject = CK_INVALID_HANDLE;
+                xResult = CKR_OK;
```

Comment:
Is this "OK" result expected?

Reply:
Yes. When no matching objects have been found, the PKCS11 standard calls for returning count=0, a non-valid handle, and status=OK.